src/data_source/storage/sql.rs

```diff
@@ -1787,6 +1789,636 @@ impl SqlStorage {
+                    Box::new(hash.to_string()),
+                    Box::new(request.num_blocks.get() as i64),
+                ],
+            ),
+        };
+
+        let result = self.query(&query, params).await.map(|row_stream| {
+            row_stream.map(|row| -> Result<BlockSummary<Types>, QueryError> {
+                let block = parse_block::<Types>(row?)?;
+                Ok(BlockSummary::try_from(block)?)
+            })
+        });
+
+        let results =
+            result.map(|res| async move { res.try_collect::<Vec<BlockSummary<Types>>>().await });
+
+        results?.await
+    }
+
+    async fn get_block_detail(
+        &self,
+        request: &explorer::data_source::BlockIdentifier<Types>,
+    ) -> QueryResult<explorer::data_source::BlockDetail<Types>> {
+        let (query, params): (String, Vec<Box<dyn ToSql + Sync + Send>>) = match request {
+            explorer::data_source::BlockIdentifier::Latest => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                        FROM header AS h
+                        JOIN payload AS p ON h.height = p.height
+                        ORDER BY h.height DESC 
+                        LIMIT 1"
+                ),
+                vec![],
+            ),
+            explorer::data_source::BlockIdentifier::Height(height) => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                        FROM header AS h
+                        JOIN payload AS p ON h.height = p.height
+                        WHERE h.height = $1
+                        ORDER BY h.height DESC 
+                        LIMIT 1"
+                ),
+                vec![Box::new(*height as i64)],
+            ),
+            explorer::data_source::BlockIdentifier::Hash(hash) => (
+                format!(
+                    "SELECT {BLOCK_COLUMNS}
+                        FROM header AS h
+                        JOIN payload AS p ON h.height = p.height
+                        WHERE h.hash = $1
+                        ORDER BY h.height DESC 
+                        LIMIT 1"
+                ),
+                vec![Box::new(hash.to_string())],
+            ),
+        };
+
+        let query_result = self.query_one(&query, params).await?;
+        let block = parse_block(query_result)?;
+
+        Ok(
+            explorer::data_source::BlockDetail::try_from(block).map_err(|e| QueryError::Error {
+                message: e.to_string(),
+            })?,
+        )
+    }
+
+    async fn get_transaction_summaries(
+        &self,
+        request: &explorer::data_source::GetTransactionSummariesRequest<Types>,
+    ) -> QueryResult<Vec<explorer::data_source::TransactionSummary<Types>>> {
+        let range = &request.range;
+        let target = &range.target;
+        let filter = &request.filter;
+
+        let (query, params, offset): (String, Vec<Box<dyn ToSql + Send + Sync>>, Option<usize>) =
+            match (target, filter) {
```

Comment:
A block stream could be a great place to start in most of these query places as they are essentially streaming from a given block.  In the majority of cases we are able to determine the starting block to a simple (block, transaction offset) combination, and then we just proceed backwards from there.  There is a case where this doesn't hold, and that's for Rollup filtering (which isn't implemented yet), but that can be handled separately if it helps to simplify the overall logic.

I'm not so certain we'd want to benefit from the existing stream implementations, as this is using the `FetchingDataSource` which will load any missing `block`s that are encountered, whereas the `Explorer` isn't attempting to have this side-effect.  It wants to just report things as they are rather than trying to proactively fetch missing data.

The hand-crafted queries in each case are primarily just a means to avoid any potential for SQL injections by eliminating any possibility of concatenation.  It also potential provides a few benefits, the first being that if we wish to pursue a prepared statement approach, these static queries could be pre-submitted to the SQL service at launch, and validated before the rest of the program engages.  This would potentially have an added benefit of slightly faster query processing.  In addition, static queries could be prefixed with an `SQL comment` that could aid in identification of the query using `SQL statement` logs.

However, the flow of the query in general is figuring out what `block`s to target, and then returning them.  This could potentially be split up into separate queries, and while we'd suffer speed due to performing multiple queries we may have gains in readability and maintainability as a result, and this seems like a fair price to pay.

Reply:
I'm actually having trouble pursuing the block stream approach as we run into some pitfalls that the current separate queries are resolving on their own.

The first is that we only pull the `block`s that have corresponding `transaction`s which prevents us from neeing to check and expand every block to see if we have the required number of transactions available.

The second is that we only pull the number of blocks we need to actually pull `N` transactions, which is what a lot of the sub-queries are doing in these cases.

As not every query takes the same number of sub-query paths it's difficult to resolve the nuances of everything with the same manner.  Usually the inner most query is utilized to determine the starting point (where applicable).  The second sub query then is used to limit the number of transactions that will be pulled up to the required `limit`.  This second sub-query can incur an addition filter for the `TransactionSummaryFilter` which helps to reduce the `block`s returned even further.  After that the `block`s are returned from only those transactions that this applies to.

Since the other variants of the queries are able to be simplified, this can make it quite difficult to reason about these further reductions that would apply.  This may require some further consideration.

Reply:
I was able to get this working such that I'm not matching on both `TransactionIdentifier` and `TransactionSummaryFilter` at the same time.   If we consider `TransactionIdentifier` has having `N` enum variants and `TransactionSummaryFilter` has having `M` variants, the previous approach would have required `N * M` hard-coded queries.

With the modified approach, I instead run a separate query for each, resulting in the need for `N + M` unique queries instead.

Reply:
This change is implemented in commit `4fd964307e8c6b25dc0c76ccb8a88f41f3973901`